service/src/main/java/bio/terra/workspace/service/resource/model/WsmResource.java

```diff
@@ -195,6 +200,88 @@ public ImmutableMap<String, String> getProperties() {
    */
   public abstract <T> T castByEnum(WsmResourceType expectedType);
 
+  /**
+   * Build a ReferencedResource clone object of this resource object. Both controlled and referenced
+   * resources can be the source of a referenced in a cloned workspace. For referenced resources,
+   * the clone is typically a copy of the reference. For controlled resources, the clone is a
+   * reference to the resource in this workspace.
+   *
+   * <p>If name or description is not supplied, then we use the one from this resource.
+   *
+   * <p>This default implementation throws CloneInstructionNotSupportedException.
+   *
+   * @param destinationWorkspaceUuid id of the destination workspace
+   * @param name optional resource name override
+   * @param description optional resource description override
+   * @return WsmResource that is the referenced resource object
+   */
+  public WsmResource buildReferencedClone(
+      UUID destinationWorkspaceUuid,
+      @Nullable UUID destinationResourceId,
+      @Nullable UUID destinationFolderId,
+      @Nullable String name,
+      @Nullable String description) {
+    throw new CloneInstructionNotSupportedException(
+        String.format(
+            "You cannot make a reference clone of a %s resource", getResourceType().name()));
+  }
+
+  protected WsmResourceFields buildCloneResourceCommonFields(
```

Comment:
This is only used for referenced dest resources (not controlled dest resources), right?

If so, put `referenced` somewhere in name.

Line 242-243 only makes sense for referenced dest resource. (For controlled dest resource, it would change COPY_RESOURCE to COPY_REFERENCE.)